test/built-ins/RegExp/named-groups/non-unicode-property-names.js

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Exotic named group names in non-Unicode RegExps
+esid: pending
+features: [regexp-named-groups]
+includes: [compareArray.js]
+---*/
+
+assert.sameValue("a", /(?<π>a)/.exec("bab").groups.π);
+assert.throws(SyntaxError, () => eval('/(?<\\u{03C0}>a)/'), "\\u{} escapes allowed only in Unicode mode");
+assert.sameValue("a", /(?<π>a)/.exec("bab").groups.\u03C0);
+assert.sameValue("a", /(?<$>a)/.exec("bab").groups.$);
+assert.sameValue("a", /(?<_>a)/.exec("bab").groups._);
+assert.sameValue("a", /(?<$𐒤>a)/.exec("bab").groups.$𐒤);
```

Comment:
If we allow 𐒤 in BMP patterns, we probably also need to allow `\ud801\udca4` (and maybe even `\u{104a4}`) for consistency reasons, because it'd strange to accept a surrogate pair only if it's written in non-escaped form. (If we want to keep it simple, we can forbid 𐒤 for the time being, and if there's any user demand, we can later revisit this decision and allow 𐒤. This won't cause any compatibility issues, because we'd only change an error into a non-error.)